### ISSUE TITLE
cupy.select produces an error

### ISSUE THREAD

**Issue by chopikus** [cupy/cupy#8453]
### Description

Error:
```
  File "/root/raytracer/src/sphere.py", line 39, in hit
    return np.select([choose_t1, choose_t2], [t1, t2], np.nan)
  File "/usr/local/lib/python3.10/dist-packages/cupy/_indexing/indexing.py", line 216, in select
    cupy.copyto(result, default)
  File "/usr/local/lib/python3.10/dist-packages/cupy/_manipulation/basic.py", line 38, in copyto
    can_cast = numpy.can_cast(src, dst.dtype, casting)
TypeError: can_cast() does not support Python ints, floats, and complex because the result used to depend on the value.
This change was part of adopting NEP 50, we may explicitly allow them again in the future.
```

### To Reproduce

```py
import cupy as np
type FloatArray = npt.NDArray[np.float64]

...
def hit(self, r: RayArray) -> FloatArray:
  Arr: TypeAlias = FloatArray
  A: Arr = directions.len_squared()
  B: Arr = -2.0 * (ocs @ directions)
  C: Arr = ocs.len_squared() - radiuses * radiuses
  D: Arr = B*B - 4*A*C
  
  cond = D >= 0 #BoolArray(fix later)
  sq = np.sqrt(np.maximum(D, 0))
  t1: Arr = np.where(cond, (-B - sq) / (2 * A), np.nan)
  t2: Arr = np.where(cond, (-B + sq) / (2 * A), np.nan)
  
  # np.isnan(t1) == np.isnan(t2) always since the condition is the same
  choose_t1 = (~np.isnan(t1)) & (t1 <= t2)
  choose_t2 = (~np.isnan(t1)) & (t1 > t2)
  
  print(type(t1))
  
  return np.select([choose_t1, choose_t2], [t1, t2], np.nan)
```

### Installation

Source (`pip install cupy`)

### Environment

```
OS                           : Linux-6.5.0-28-generic-x86_64-with-glibc2.35
Python Version               : 3.10.12
CuPy Version                 : 13.2.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 2.0.1
SciPy Version                : None
Cython Build Version         : 0.29.37
Cython Runtime Version       : None
CUDA Root                    : /usr/local/cuda
nvcc PATH                    : /usr/local/cuda/bin/nvcc
CUDA Build Version           : 12040
CUDA Driver Version          : 12040
CUDA Runtime Version         : 12040 (linked to CuPy) / 12040 (locally installed)
cuBLAS Version               : (available)
cuFFT Version                : 11201
cuRAND Version               : 10305
cuSOLVER Version             : (11, 6, 1)
cuSPARSE Version             : (available)
NVRTC Version                : (12, 4)
Thrust Version               : 200200
CUB Build Version            : 200200
Jitify Build Version         : <unknown>
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : 22105
NCCL Runtime Version         : 22105
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce RTX 2070 SUPER
Device 0 Compute Capability  : 75
Device 0 PCI Bus ID          : 0000:03:00.0

```

**Comment by seberg** [comment]
This is already fixed for CuPy 13.3 (not yet released), you can downgrade NumPy to `numpy<2` to avoid it.

**Comment by kmaehashi** [comment]
Closing as the fix is already in. We plan to ship the fix this month. Thanks for reporting @chopikus!
